Automattic/wp-calypso
# Checkout: Remove undocumented.transactions function

#### Changes proposed in this Pull Request

As part of the goal to remove the `wpcom.undocumented` module (see 429-gh-Automattic/payments-shilling), this PR modifies checkout to call the `/me/transactions` endpoint directly. This is the endpoint used to submit a purchase by all payment methods except for PayPal.

(This PR isn't as big as it looks. Mostly it's just updating the tests to mock the actual endpoint calls instead of being able to mock the `undocumented` function.)

#### Testing instructions

- Add a product to your cart and visit checkout.
- Chose any payment method other than PayPal and submit the purchase.
- Verify that the purchase is successful.
